### ISSUE TITLE
🐛 [firebase_auth]  MissingPluginException(No implementation found for method Auth#registerIdTokenListener) and more

### ISSUE THREAD

**Issue by remcova** [firebase/flutterfire#11234]
## Bug report

I'm using firebase_auth 4.6.3 and every time when I sign-in with my Google account in my Flutter desktop application, I get the following error:
```
[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MissingPluginException(No implementation found for method Auth#registerIdTokenListener on channel plugins.flutter.io/firebase_auth)
#0      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:308:7)
<asynchronous suspension>

[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MissingPluginException(No implementation found for method Auth#registerAuthStateListener on channel plugins.flutter.io/firebase_auth)
#0      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:308:7)
<asynchronous suspension>

[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MissingPluginException(No implementation found for method Auth#signInWithCredential on channel plugins.flutter.io/firebase_auth)
#0      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:308:7)
<asynchronous suspension>
#1      MethodChannel.invokeMapMethod (package:flutter/src/services/platform_channel.dart:510:43)
<asynchronous suspension>
#2      MethodChannelFirebaseAuth.signInWithCredential (package:firebase_auth_platform_interface/src/method_channel/method_channel_firebase_auth.dart:459:12)
<asynchronous suspension>
```

### Steps to reproduce
Install firebase_auth 4.6.3 and use FirebaseAuth.instance.signInWithCredential();

### Expected behavior
The functions should be present?

---

### Flutter doctor

<details><summary>Click To Expand</summary>

```
[√] Flutter (Channel stable, 3.10.5, on Microsoft Windows [Version 10.0.19045.3086], locale nl-NL)
[√] Windows Version (Installed version of Windows is version 10 or higher)
[X] Android toolchain - develop for Android devices
    X Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.

[√] Chrome - develop for the web
[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.6.4)
[!] Android Studio (not installed)
[√] VS Code (version 1.79.2)
[√] Connected device (3 available)
[√] Network resources
```

</details>

---

### Flutter dependencies

<details><summary>Click To Expand</summary>

```
dependencies:
- audioplayers 4.1.0 [audioplayers_android audioplayers_darwin audioplayers_linux audioplayers_platform_interface audioplayers_web audioplayers_windows file flutter http meta path_provider synchronized uuid]
- base32 2.1.3
- cupertino_icons 1.0.5
- desktop_drop 0.4.1 [flutter flutter_web_plugins cross_file]
- desktop_webview_auth 0.0.9 [crypto flutter http]
- desktop_window 0.4.0 [flutter]
- dio 4.0.6 [http_parser path]
- email_validator 2.1.17
- extension_google_sign_in_as_googleapis_auth 2.0.10 [flutter google_sign_in googleapis_auth http meta]
- ffmpeg_cli 0.1.0 [json_annotation logging path_provider]
- ffmpeg_kit_flutter_https_gpl 5.1.0 [ffmpeg_kit_flutter_platform_interface flutter]
- file_picker 5.3.2 [flutter flutter_web_plugins flutter_plugin_android_lifecycle plugin_platform_interface ffi path win32]
- firebase_auth 4.6.3 [firebase_auth_platform_interface firebase_auth_web firebase_core firebase_core_platform_interface flutter meta]
- firebase_core 2.14.0 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_core_desktop 1.0.2 [firebase_core_dart firebase_core_platform_interface flutter meta]
- flash 3.0.5+1 [flutter]
- flutter 0.0.0 [characters collection js material_color_utilities meta vector_math sky_engine]
- flutter_signin_button 2.0.0 [flutter font_awesome_flutter]
- fluttericon 2.0.0 [flutter]
- google_sign_in 6.1.4 [flutter google_sign_in_android google_sign_in_ios google_sign_in_platform_interface google_sign_in_web]
- googleapis 11.2.0 [_discoveryapis_commons http]
- googleapis_auth 1.4.1 [args crypto http http_parser js]
- image 4.0.17 [archive meta xml]
- otp 3.1.4 [base32 crypto logging timezone]
- path 1.8.3
- percent_indicator 4.2.3 [flutter]
- provider 6.0.5 [collection flutter nested]
- qr_flutter 4.1.0 [flutter qr]
- settings_ui 2.0.2 [flutter]
- shared_preferences 2.2.0 [flutter shared_preferences_android shared_preferences_foundation shared_preferences_linux shared_preferences_platform_interface shared_preferences_web shared_preferences_windows]
- top_snackbar_flutter 3.1.0 [flutter]
- url_launcher 6.1.11 [flutter url_launcher_android url_launcher_ios url_launcher_linux url_launcher_macos url_launcher_platform_interface url_launcher_web url_launcher_windows]
- window_manager 0.3.5 [flutter path screen_retriever]
- yt 2.0.6+9 [args csv dio fling_pickle googleapis_auth http http_parser json_annotation loggy process_run retrofit stemmer universal_io yaml] 

dev dependencies:
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters collection js matcher material_color_utilities meta source_span stream_channel string_scanner term_glyph]
- icons_launcher 2.1.2 [args image path universal_io yaml]

dependency overrides:
- desktop_webview_auth 0.0.9 [crypto flutter http]

transitive dependencies:
- _discoveryapis_commons 1.0.6 [http http_parser meta]
- _fe_analyzer_shared 61.0.0 [meta]
- _flutterfire_internals 1.3.3 [collection firebase_core firebase_core_platform_interface flutter meta]
- analyzer 5.13.0 [_fe_analyzer_shared collection convert crypto glob meta package_config path pub_semver source_span watcher yaml]
- archive 3.3.7 [crypto path pointycastle]
- args 2.4.2
- async 2.11.0 [collection meta]
- audioplayers_android 3.0.2 [audioplayers_platform_interface flutter]
- audioplayers_darwin 4.1.0 [audioplayers_platform_interface flutter]
- audioplayers_linux 2.1.0 [audioplayers_platform_interface flutter]
- audioplayers_platform_interface 5.0.1 [flutter meta plugin_platform_interface]
- audioplayers_web 3.1.0 [audioplayers_platform_interface flutter flutter_web_plugins js]
- audioplayers_windows 2.0.2 [audioplayers_platform_interface flutter]
- boolean_selector 2.1.1 [source_span string_scanner]
- characters 1.3.0
- charcode 1.3.1
- clock 1.1.1
- collection 1.17.1
- convert 3.1.1 [typed_data]
- coverage 1.6.3 [args logging package_config path source_maps stack_trace vm_service]
- cross_file 0.3.3+4 [js meta]
- crypto 3.0.3 [typed_data]
- csv 5.0.2
- fake_async 1.3.1 [clock collection]
- ffi 2.0.2
- ffmpeg_kit_flutter_platform_interface 0.2.1 [flutter plugin_platform_interface]
- file 6.1.4 [meta path]
- firebase_auth_platform_interface 6.15.3 [_flutterfire_internals collection firebase_core flutter meta plugin_platform_interface]
- firebase_auth_web 5.5.3 [firebase_auth_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins http_parser js meta]   
- firebase_core_dart 1.0.1 [collection meta]
- firebase_core_platform_interface 4.8.0 [collection flutter flutter_test meta plugin_platform_interface]
- firebase_core_web 2.6.0 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- fling_pickle 0.4.3
- flutter_plugin_android_lifecycle 2.0.15 [flutter]
- flutter_web_plugins 0.0.0 [flutter js characters collection material_color_utilities meta vector_math]
- font_awesome_flutter 9.2.0 [flutter]
- frontend_server_client 3.2.0 [async path]
- glob 2.1.2 [async collection file path string_scanner]
- google_identity_services_web 0.2.1 [js meta]
- google_sign_in_android 6.1.18 [flutter google_sign_in_platform_interface]
- google_sign_in_ios 5.6.2 [flutter google_sign_in_platform_interface]
- google_sign_in_platform_interface 2.4.1 [flutter plugin_platform_interface quiver]
- google_sign_in_web 0.12.0+2 [flutter flutter_web_plugins google_identity_services_web google_sign_in_platform_interface http js]
- http 0.13.6 [async http_parser meta]
- http_multi_server 3.2.1 [async]
- http_parser 4.0.2 [collection source_span string_scanner typed_data]
- io 1.0.4 [meta path string_scanner]
- js 0.6.7 [meta]
- json_annotation 4.8.1 [meta]
- logging 1.2.0
- loggy 2.0.3 [stack_trace]
- matcher 0.12.15 [async meta stack_trace term_glyph test_api]
- material_color_utilities 0.2.0
- meta 1.9.1
- mime 1.0.4
- nested 1.0.0 [flutter]
- node_preamble 2.0.2
- package_config 2.1.0 [path]
- path_provider 2.0.15 [flutter path_provider_android path_provider_foundation path_provider_linux path_provider_platform_interface path_provider_windows]
- path_provider_android 2.0.27 [flutter path_provider_platform_interface]
- path_provider_foundation 2.2.3 [flutter path_provider_platform_interface]
- path_provider_linux 2.1.11 [ffi flutter path path_provider_platform_interface xdg_directories]
- path_provider_platform_interface 2.0.6 [flutter platform plugin_platform_interface]
- path_provider_windows 2.1.7 [ffi flutter path path_provider_platform_interface win32]
- petitparser 5.4.0 [meta]
- platform 3.1.0
- plugin_platform_interface 2.1.4 [meta]
- pointycastle 3.7.3 [collection convert js]
- pool 1.5.1 [async stack_trace]
- process 4.2.4 [file path platform]
- process_run 0.12.5+3 [path collection charcode string_scanner yaml meta args pub_semver synchronized]
- pub_semver 2.1.4 [collection meta]
- qr 3.0.1 [meta]
- quiver 3.2.1 [matcher]
- retrofit 3.3.1 [dio meta]
- screen_retriever 0.1.6 [flutter]
- shared_preferences_android 2.2.0 [flutter shared_preferences_platform_interface]
- shared_preferences_foundation 2.3.0 [flutter shared_preferences_platform_interface]
- shared_preferences_linux 2.3.0 [file flutter path path_provider_linux path_provider_platform_interface shared_preferences_platform_interface]- shared_preferences_platform_interface 2.3.0 [flutter plugin_platform_interface]
- shared_preferences_web 2.2.0 [flutter flutter_web_plugins shared_preferences_platform_interface]
- shared_preferences_windows 2.3.0 [file flutter path path_provider_platform_interface path_provider_windows shared_preferences_platform_interface]
- shelf 1.4.1 [async collection http_parser path stack_trace stream_channel]
- shelf_packages_handler 3.0.2 [path shelf shelf_static]
- shelf_static 1.1.2 [convert http_parser mime path shelf]
- shelf_web_socket 1.0.4 [shelf stream_channel web_socket_channel]
- sky_engine 0.0.99
- source_map_stack_trace 2.1.1 [path source_maps stack_trace]
- source_maps 0.10.12 [source_span]
- source_span 1.9.1 [collection path term_glyph]
- stack_trace 1.11.0 [path]
- stemmer 2.2.0 [test]
- stream_channel 2.1.1 [async]
- string_scanner 1.2.0 [source_span]
- synchronized 3.1.0
- term_glyph 1.2.1
- test 1.24.1 [analyzer async boolean_selector collection coverage http_multi_server io js node_preamble package_config path pool shelf shelf_packages_handler shelf_static shelf_web_socket source_span stack_trace stream_channel typed_data web_socket_channel webkit_inspection_protocol yaml test_api test_core matcher]
- test_api 0.5.1 [async boolean_selector collection meta source_span stack_trace stream_channel string_scanner term_glyph matcher]
- test_core 0.5.1 [analyzer async args boolean_selector collection coverage frontend_server_client glob io meta package_config path pool source_map_stack_trace source_maps source_span stack_trace stream_channel vm_service yaml matcher test_api]
- timezone 0.9.2 [path]
- typed_data 1.3.2 [collection]
- universal_io 2.2.2 [collection meta typed_data]
- url_launcher_android 6.0.36 [flutter url_launcher_platform_interface]
- url_launcher_ios 6.1.4 [flutter url_launcher_platform_interface]
- url_launcher_linux 3.0.5 [flutter url_launcher_platform_interface]
- url_launcher_macos 3.0.5 [flutter url_launcher_platform_interface]
- url_launcher_platform_interface 2.1.3 [flutter plugin_platform_interface]
- url_launcher_web 2.0.17 [flutter flutter_web_plugins url_launcher_platform_interface]
- url_launcher_windows 3.0.6 [flutter url_launcher_platform_interface]
- uuid 3.0.7 [crypto]
- vector_math 2.1.4
- vm_service 11.7.2
- watcher 1.1.0 [async path]
- web_socket_channel 2.4.0 [async crypto stream_channel]
- webkit_inspection_protocol 1.2.0 [logging]
- win32 5.0.5 [ffi]
- xdg_directories 1.0.0 [meta path process]
- xml 6.3.0 [collection meta petitparser]
- yaml 3.1.2 [collection source_span string_scanner]
```

</details>

---

**Comment by darshankawar** [comment]
@remcova 
Since you seem to be trying to run `firebase_auth` on Windows desktop platform, can you try https://pub.dev/packages/firebase_auth_desktop and see if it helps to resolve the issue ?

**Comment by remcova** [comment]
@darshankawar Thank you for your quick reply. When I try `firebase_auth_desktop`, I get the following error:
```
../../../AppData/Local/Pub/Cache/hosted/pub.dev/firebase_core-1.20.0/lib/src/firebase_app.dart(18,25): error G75B77105: Member not found: 'FirebaseAppPlatform.verifyExtends'. [C:\Users\rvanakker\Documents\GitHub\project\build\windows\flutter\flutter_assemble.vcxproj]
../../../AppData/Local/Pub/Cache/hosted/pub.dev/firebase_auth_platform_interface-6.10.1/lib/src/action_code_info.dart(65,15): error GE5CFE876: The method 'FallThroughError' isn't defined for the class 'ActionCodeInfo'. [C:\Users\rvanakker\Documents\GitHub\project\build\windows\flutter\flutter_assemble.vcxproj]
../../../AppData/Local/Pub/Cache/hosted/pub.dev/firebase_auth-3.6.2/lib/src/firebase_auth.dart(623,25): error GE5CFE876: The method 'signInWithAuthProvider' isn't defined for the class 'FirebaseAuthPlatform'. [C:\Users\rvanakker\Documents\GitHub\project\build\windows\flutter\flutter_assemble.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(248,5): error MSB8066: Custom build for 'C:\Users\rvanakker\Documents\GitHub\project\build\windows\CMakeFiles\777f883dcc32a51bbfdbaba483a84dab\flutter_windows.dll.rule;C:\Users\rvanakker\Documents\GitHub\project\build\windows\CMakeFiles\d01b44a161dbbf5964884bd1e3831c47\flutter_assemble.rule' exited with code 1. [C:\Users\rvanakker\Documents\GitHub\project\build\windows\flutter\flutter_assemble.vcxproj]
Exception: Build process failed.
```

**Comment by darshankawar** [comment]
@remcova 
Regarding the error you posted above, `Member not found: `FirebaseAppPlatform.verifyExtends`,  it has been fixed already as part of https://github.com/firebase/flutterfire/issues/9806 issue, so make sure you are using latest versions of all flutterfire packages as also mentioned in that https://github.com/firebase/flutterfire/issues/9806#issuecomment-1319642334.

**Comment by remcova** [comment]
> @remcova Regarding the error you posted above, `Member not found: FirebaseAppPlatform.verifyExtends`, it has been fixed already as part of #9806 issue, so make sure you are using latest versions of all flutterfire packages as also mentioned in that [#9806 (comment)](https://github.com/firebase/flutterfire/issues/9806#issuecomment-1319642334).

Updating all the Flutterfire packages works! but I want to mention that a few packages are outdated. I had to update these myself. Thank you for your time.

**Comment by darshankawar** [comment]
Closing as resolved.
